src/docker/composeappengine.cc

```diff
@@ -107,6 +107,19 @@ AppEngine::Result ComposeAppEngine::run(const App& app) {
     return {false, "App dir doesn't exist, cannot install App that hasn't been fetched"};
   }
 
+  if (!isFetched(app)) {
+    // This situation can occur only if the switch from restorable to compose apps
+    // happened or App was pulled manually by running `docker-compose` command.
+    // So, the app directory with the compose content is present while the `.meta` directory
+    // and the `.meta/.version` files are missing.
+    // Therefore, we assume that App has been already pulled and set its state correspondingly.
+    LOG_WARNING << "Trying to run App if it's not fetched or failure to get its status...";
+    // make sure the app meta dir exists
+    boost::filesystem::create_directories(appRoot(app) / MetaDir);
+    // assume that the app state is "pulled"
+     AppState(app, appRoot(app), true).setState(AppState::State::kPulled);
```

Comment:
This indentation looks off? Guessing clang-tidy will spot that. 

Nitpick - but the `if it's` part of the log_warning message confuses me.

Reply:
I fixed the indentation.

>Nitpick - but the if it's part of the log_warning message confuses me.
English grammar or the content.

Reply:
not sure - maybe both. I think you are trying to say something like "The App appears to have either been fetched manually or an unexpected error occurred reading its status"?

Reply:
>The App appears to have either been fetched manually

This is an assumption and one of the possible reasons.
So, I am trying to convey the state/situation not what might led to it.

The state can be either:
1. No compose content at all in the app dir
2. compose content is there, just the state file is missing.
3. compose content and the state file is there, but a failure to read it occurs.

Reply:
maybe you just need to re-word the first half then `Trying to run App if it is not fetched`.

Reply:
Maybe `App state is unknown, assuming it's pulled and trying to run it...`.
The error message about failure to read it should appear before it, the `isFetched()` call should print it.

Reply:
sounds good. thanks

Reply:
Updated.